tikv-client/src/main/java/com/pingcap/tikv/codec/TableCodec.java

```diff
@@ -66,15 +83,32 @@ public static Handle decodeHandle(byte[] value, boolean isCommonHandle) {
   }
 
   /* only for unique index */
-  public static byte[] genIndexValueForClusteredIndexVersion1(TiIndexInfo index, Handle handle) {
+  public static Handle decodeHandleInUniqueIndexValue(byte[] value, boolean isCommonHandle) {
+    if (!isCommonHandle) {
+      return new IntHandle(new CodecDataInput(value).readLong());
+    }
+    CodecDataInput codecDataInput = new CodecDataInput(value);
+    if (getIndexVersion(value) == 1) {
+      IndexValueSegments segments = splitIndexValueForClusteredIndexVersion1(codecDataInput);
+      return new CommonHandle(segments.commonHandle);
+    }
+    int handleLen = ((int) value[2]) << 8 + value[3];
+    byte[] encode = Arrays.copyOfRange(value, 4, handleLen + 4);
+    return new CommonHandle(encode);
+  }
+
+  /* only for unique index */
+  public static byte[] genIndexValueForClusteredIndexVersion1(
+      TiIndexInfo index, Handle handle, boolean distinct) {
     CodecDataOutput cdo = new CodecDataOutput();
     cdo.writeByte(0);
     cdo.writeByte(IndexVersionFlag);
     cdo.writeByte(1);
 
     assert (index.isUnique());
-    encodeCommonHandle(cdo, handle);
-
+    if (distinct) {
```

Comment:
why judge it

Reply:
because only the distinct index key needs handle values in the index value.

Reply:
> why judge it

because only the distinct index key needs handle values in the index value.